snap/snapcraft.yaml

```diff
@@ -32,13 +32,16 @@ parts:
     plugin: go
     source: https://github.com/hashicorp/terraform
     source-type: git
-    source-tag: "v$SNAPCRAFT_PROJECT_VERSION"
+    override-pull: |
+      VERSION="$(craftctl get version)"
+      git clone -b "v$VERSION" https://github.com/hashicorp/terraform $CRAFT_PART_SRC
+    build-snaps: [go]
```

Comment:
@sergiusens the `go` plugin doesn't seem to install `go` by default unless you specify this  - is this a bug?